nengo/networks/assoc_mem.py

```diff
@@ -0,0 +1,436 @@
+                nengo.Connection(e, self.elem_output[i],
+                                 function=self.filt_step_func, synapse=None)
+
+            # Configure associative memory to be inhibitable
+            if inhibitable:
+                # Input node for inhibitory gating signal (if enabled)
+                self.inhibit = nengo.Node(size_in=1, label="inhibit")
+                nengo.Connection(self.inhibit, self.elem_input,
+                                 transform=-np.ones((N, 1)) * 1.5,
+                                 synapse=None)
+                # Note: We can use decoded connection here because all the
+                # encoding vectors are [1]
+            else:
+                self.inhibit = None
+
+            # Configure utilities output
+            self.utilities = self.elem_output
+
+            c = nengo.Connection(self.elem_output, self.output,
+                                 transform=output_vectors.T, synapse=None)
+
+            # Add the output connection to the output connection list
+            self.out_conns.append(c)
+
+    """Function to add another input to the associative memory network.
+
+    Parameters
+    ----------
+    name: string
+        Name to use for the input node. This name will be used as the name of
+        attribute for the associative memory network.
+
+    input_vectors: array_like
+        The list of vectors to be compared against.
+    input_scales: array_list, optional
+        Scaling factor to apply on each of the input vectors. Note that it
+        is possible to scale each vector independently.
+    """
+    @with_self
+    def add_input(self, name, input_vectors, input_scales=1.0):
+        # Handle different vocabulary types
+        if is_iterable(input_vectors):
+            input_vectors = np.matrix(input_vectors)
+
+        # Handle possible different input_scale values for each
+        # element in the associative memory
+        if not is_iterable(input_scales):
+            input_scales = np.array([input_scales] * input_vectors.shape[0],
+                                    ndmin=2)
+        else:
+            input_scales = np.array(input_scales, ndmin=2)
+        if input_scales.shape[1] != input_vectors.shape[0]:
+            raise ValueError(
+                'Number of input_scale values do not match number of input '
+                'vectors. Got: %d, expected %d.' %
+                (input_scales.shape[1], input_vectors.shape[0]))
+
+        input = nengo.Node(size_in=input_vectors.shape[1], label=name)
+
+        if hasattr(self, name):
+            raise NameError('Name "%s" already exists as a node in the'
+                            'associative memory.')
+        else:
+            setattr(self, name, input)
+
+        nengo.Connection(input, self.elem_input,
+                         synapse=None,
+                         transform=np.multiply(input_vectors, input_scales.T))
+
+    """Function to add another input to the associative memory network.
```

Comment:
Should say 'another output'. Also I would remove the "Function to". It seems redundant as it should be clear that it is a function (also for the `add_input` function).